recipes/wasmtime/all/conanfile.py

```diff
@@ -0,0 +1,103 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration, ConanException
+import os
+
+
+class WasmtimeConan(ConanFile):
+    name = 'wasmtime'
+    homepage = 'https://github.com/bytecodealliance/wasmtime'
+    license = 'Apache-2.0'
+    url = 'https://github.com/conan-io/conan-center-index'
+    description = "Standalone JIT-style runtime for WebAssembly, using Cranelift"
+    topics = ("webassembly", "wasm", "wasi")
+    settings = "os", "compiler", "arch"
+    options = {
+        "shared": [True, False],
+        'fPIC': [True],
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+    }
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "apple-clang": "9.4",
+            "clang": "3.3",
+            "gcc": "4.9.4"
+        }
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        compiler = self.settings.compiler
+        min_version = self._minimum_compilers_version[str(compiler)]
+        try:
+            if tools.Version(compiler.version) < min_version:
+                msg = (
+                    "{} requires C{} features which are not supported by compiler {} {} !!"
+                ).format(self.name, self._minimum_cpp_standard, compiler, compiler.version)
+                raise ConanInvalidConfiguration(msg)
+        except KeyError:
+            msg = (
+                "{} recipe lacks information about the {} compiler, "
+                "support for the required C++{} features is assumed"
+            ).format(self.name, compiler, self._minimum_cpp_standard)
+            self.output.warn(msg)
+
+        if (not (self.version in self.conan_data["sources"]) or
+            not (str(self.settings.os) in self.conan_data["sources"][self.version]) or
+            not (str(self.settings.arch) in self.conan_data["sources"][self.version][str(self.settings.os)] ) ):
+            raise ConanInvalidConfiguration("Binaries for this combination of architecture/version/os not available")
```

Comment:
I think this is more "pythonic"
```suggestion
        try:
            self.conan_data["sources][self.version][str(self.settings.os)]
        except KeyError:
            raise ConanInvalidConfiguration("Binaries for this combination of architecture/version/os not available")
```

Reply:
Done